src/legacy/legacy-default-xpring-client.ts

```diff
@@ -42,9 +40,18 @@ class LegacyDefaultXpringClient implements XpringClientDecorator {
    */
   public static defaultXpringClientWithEndpoint(
     grpcURL: string,
+    forceHttp = false,
   ): LegacyDefaultXpringClient {
-    const grpcClient = new LegacyGRPCNetworkClient(grpcURL)
-    return new LegacyDefaultXpringClient(grpcClient)
+    if (
+      typeof process !== 'undefined' &&
+      process.release.name === 'node' &&
+      !forceHttp
```

Comment:
Can we throw this logic into a helper function somewhere (without the `forceHTTP` probably)

Reply:
Okay did this